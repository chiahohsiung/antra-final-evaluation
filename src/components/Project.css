.project-card {
  width: 300px;
  height: 300px;
  margin: 20px;
}

.project-card h3 {
  padding-top: 20px;
}
