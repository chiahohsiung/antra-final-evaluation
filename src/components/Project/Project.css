.project-card {
  width: 300px;
  max-height: 300px;
  overflow: auto;
  margin: 20px;
  flex-grow: 1;
}

.project-card h3 {
  padding-top: 20px;
}
