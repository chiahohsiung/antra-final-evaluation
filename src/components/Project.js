import React from 'react'
import './Project.css'
function Project(props) {
  const {title, detail, titleClass } = props;
  return (
    <div className="project-card text-center">
      <div className={titleClass}>
        <h3>{title}</h3>
        <p>{detail}</p>

        <div className="project-more-info">
          <a className="btn btn-default btn-border" href="#">More</a>
        </div>

      </div>
    </div>
  )
}

export default Project;