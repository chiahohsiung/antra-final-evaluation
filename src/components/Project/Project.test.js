import React from 'react'
import { render } from '@testing-library/react'
import Project from './Project'

// shallow test 
describe('Project', () => {
  it('should match snapshot', () => {
    const title = 'test-title'
    const detail = 'test-detail'
    const titleClass = 'luvtalk-project'
    const { asFragment } = render(
      <Project
        title={title}
        detail={detail}
        titleClass={titleClass}
      />
    ); // like a DOM
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render project based on props', () => {
    const title = 'test-title'
    const detail = 'test-detail'
    const titleClass = 'luvtalk-project'
    const wrapper = render(
      <Project
        title={title}
        detail={detail}
        titleClass={titleClass}
      />
    ); // like a DOM

    const projectCardTitle = wrapper.queryByText('test-title');
    expect(projectCardTitle).toBeInTheDocument();
    expect(projectCardTitle.tagName).toBe('H3');
    const projectCardDiv = wrapper.queryByText('test-title').closest('div');
    expect(projectCardDiv).toBeInTheDocument();
    expect(projectCardDiv).toHaveClass('luvtalk-project')
    // expect(summaryCardImage)
  })
})