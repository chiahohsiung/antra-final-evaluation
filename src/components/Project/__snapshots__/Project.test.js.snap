// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="project-card text-center"
  >
    <div
      class="luvtalk-project"
    >
      <h3>
        test-title
      </h3>
      <p>
        test-detail
      </p>
      <div
        class="project-more-info"
      >
        <a
          class="btn btn-default btn-border"
          href="#"
        >
          More
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
